pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -22,6 +22,10 @@ type LifecycleProcessor struct {
 	DetectLambdaLibrary func() bool
 }
 
+// InferredSpansEnabled tells us if the Env Vars are enabled
+// for inferred spans to be created
+var InferredSpansEnabled, inferredSpan = initializeInferredSpan()
```

Comment:
I think we can avoid `InferredSpansEnabled` as a global variable by carrying it in the `LifecycleProcessor` and init it in `cmd/serverless/main.go`